src/main/java/rx/schedulers/ExecutorScheduler.java

```diff
@@ -73,11 +66,15 @@ public Subscription schedule(Action0 action) {
             if (isUnsubscribed()) {
                 return Subscriptions.unsubscribed();
             }
-            ExecutorAction ea = new ExecutorAction(action, tasks);
+            ScheduledAction ea = new ScheduledAction(action, tasks);
             tasks.add(ea);
             queue.offer(ea);
             if (wip.getAndIncrement() == 0) {
                 try {
+                    // note that since we schedule the emission of potentially multiple tasks
+                    // there is no clear way to cancel this schedule from individual tasks
+                    // so even if executor is an ExecutorService, we can't associate the future
+                    // returned by submit() with any particular ScheduledAction
```

Comment:
This is the scenario: I submit 2 tasks concurrently. The first one will start the schedule with ```this```. Now let's assume the executor is busy so ```this``` is waiting in the pool's queue. If I cancel the 2 tasks, in theory, there is no need to run ```this``` anymore but to cancel it, one needs very complicated accounting. So in other terms, the best we can do here is to do the emission loop and check if the particular ScheduledAction is unsubscribed or not. The downside is the long retention of such tasks.